### ISSUE TITLE
systemd.mount - add support to mount / via .mount unit files.

### ISSUE THREAD

**Issue by ghost** [systemd/systemd#7945]
### Submission type

 - Request for enhancement (RFE)

### systemd version the issue has been seen with

systemctl --version
systemd 236
+PAM +AUDIT +SELINUX +IMA +APPARMOR +SMACK +SYSVINIT +UTMP +LIBCRYPTSETUP +GCRYPT +GNUTLS +ACL +XZ +LZ4 +SECCOMP +BLKID +ELFUTILS +KMOD -IDN2 +IDN default-hierarchy=hybrid

### Used distribution

cat /etc/debian_version
buster/sid

### Request for enhancement (RFE)

after reading the documentation of systemd.mount i came to the conclusion that mounting / via .mount unit files is impossible.

RFE: add support to mount / via .mount unit files.

**Comment by poettering** [comment]
Well, what precisely are you suggesting? The thing is that systemd itself runs of / of course, hence to run systemd you need / mounted, hence it can't be systemd that mounts /...

I am really not grokking what you are suggesting here?

**Comment by mog71** [comment]
i am not going to leave a single summer wheel on my car when winter comes.

my goal was to transfer all of my /etc/fstab statements into systemd .mount units. the same like getting all /etc/init.d scripts into systemd .service units. or getting all cronjob entries into systemd .timer units.

i dont see why one unit type should be an exception from all other unit types.

one of the first things PID1 has been doing in the past is mounting the root filesystem read-write. i expect that systemd is doing the same, so there should be already some kind of mount root filesystem unit available. i would like to have the ability to modify this unit as usual and get rid of my last /etc/fstab entry.

it would also reduce (very) slightly my configuration management tools (chef/puppet/ansible) complexity, because i dont have to make exceptions for parameters and attributes when using /etc/fstab or .mount unit files.



Am 22.01.18 um 10:56 schrieb Lennart Poettering:

**Comment by poettering** [comment]
> i dont see why one unit type should be an exception from all other unit types.

well, the root mount is different from all others, because somebody else has to mount it for the host systemd and only after it did that will invoke the host systemd itself. For example the kernel itself or the initrd, based on the kernel cmdline configuration will do this for you. And your initrd genration script or boot loader script will usually look at /etc/fstab to figure out what to mount as /, but the host systemd really doesn't need that at all anymore, it won't use the mount source your list there ever... If you patch your boot loader scripts or initrd generation scripts the right way and teach them to not bother with /etc/fstab then all should be good. But that's really outside of systemd's own scope...

In systemd there's indeed "systemd-remountfs.service" which will apply whatever is configured in /etc/fstab for the root fs onto the root fs, but that's primarily done to make sure that whatever is listed in etc/fstab counts, if it exists, not because there was much necessity in having it there...

Also note that if you run systemd in your initrd, then it will automatically discover the root partition for you (under some conditions, see systemd-gpt-auto-generator(8) for details) so that you don't need to list anything anywhere, the root fs will be found automatically...

Long story short: we don't provide this functionality because we can't really: the initrd/boot loader scripts are responsible for mounting things, and if they can't read unit files (and only /etc/fstab), then this would have to be fixed there. However, I personally think that this shouldn't be configured anywhere in fact, and initrd should just rely on systemd-gpt-auto-generator(8), and /etc/fstab should hence be removed entirely)

Hope this makes sense?

**Comment by UVV-gh** [comment]
I accidentally stumbled upon this thread when I was looking for solution for my issue.
I wanted to have / mounted second time as a lower layer of overlay fs, i.e. something like 
`mount -t overlay overlay -olowerdir=/,upperdir=/tmp/rw,workdir=/tmp/rootfs_workdir /`

How would I do that?

**Comment by poettering** [comment]
@UVV-gh this is a bug and feature request tracker. Not a support forum. Use the systemd-devel mailing list if you have questoins regarding use of systemd. Thank you for your understanding.
